posthog/queries/trends/lifecycle_actors.py

```diff
@@ -0,0 +1,44 @@
+from typing import Dict, Optional, Tuple
+
+from posthog.queries.actor_base_query import ActorBaseQuery
+from posthog.queries.trends.lifecycle import LifecycleEventQuery
+from posthog.models.filters.lifecycle_filter import LifecycleFilter
+from rest_framework.exceptions import ValidationError
+
+
+class LifecycleActors(ActorBaseQuery):
+    event_query_class = LifecycleEventQuery
+    _filter: LifecycleFilter
+
+    QUERY_TYPE = "lifecycle"
+
+    def actor_query(self, limit_actors: Optional[bool] = True) -> Tuple[str, Dict]:
+        events_query, event_params = self.event_query_class(
+            filter=self._filter,
+            team=self._team,
+            person_on_events_mode=self._team.person_on_events_mode,
+        ).get_query()
+
+        lifecycle_type = self._filter.lifecycle_type
+        if lifecycle_type is None:
+            raise ValidationError("Must include lifecycle type")
+        target_date = self._filter.target_date
+        if target_date is None:
+            raise ValidationError("Must include specified target date")
+
+        return (
```

Comment:
when you reconcile the two, you'll find that you can use `LIFECYCLE_QUERY_PEOPLE` sql from https://github.com/PostHog/posthog/blob/master/posthog/queries/trends/lifecycle.py#L61 here

Reply:
instead of re-writing the SQL here